### PR TITLE
SW-5043 Participants CRUD API

### PR DESCRIPTION
**Pull request by sgrimm**
Add CRUD endpoints for participant management.

### CODE REVIEW

**Comment by sgrimm** [comment]
* **#1814** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1814?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#1813** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1813?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1812** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1812?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @sgrimm and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/main/kotlin/com/terraformation/backend/accelerator/api/ParticipantsController.kt [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+  @ApiResponse200
+  @ApiResponse404
+  @GetMapping("/{participantId}")
+  @Operation(summary = "Gets information about a participant and its assigned projects.")
+  fun getParticipant(@PathVariable participantId: ParticipantId): GetParticipantResponsePayload {
+    val model = participantStore.fetchOneById(participantId)
+
+    return makeGetResponse(model)
+  }
+
```
Comment (tommylau523) [inline src/main/kotlin/com/terraformation/backend/accelerator/api/ParticipantsController.kt:76]: will we also want a GET / for retrieving a list of participants? Or maybe we can use search for that.

**Review by tommylau523 — APPROVED** [review]

File: src/main/kotlin/com/terraformation/backend/accelerator/api/ParticipantsController.kt [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    val organizationId: OrganizationId,
+    val organizationName: String,
+    val projectId: ProjectId,
+    val projectName: String,
+)
+
+data class ParticipantPayload(
+    val cohortId: CohortId?,
+    val cohortName: String?,
+    val id: ParticipantId,
```
Comment (tommylau523) [inline src/main/kotlin/com/terraformation/backend/accelerator/api/ParticipantsController.kt:132]: can we add phase to this payload? I think everywhere in the front-end that uses cohort name, phase is also displayed.

File: src/main/kotlin/com/terraformation/backend/accelerator/api/ParticipantsController.kt (hunk 1 above)
Comment (sgrimm) [reply src/main/kotlin/com/terraformation/backend/accelerator/api/ParticipantsController.kt:76]: We can add one if we need it, but the only places I saw where we would need to get lists of participants were either table views with CSV export (which means they need to use search anyway) or participant selection that's covered by `GET /api/v1/accelerator/organizations`.

**Comment by sgrimm** [comment]
### Merge activity

* **Mar 27, 1:25 PM EDT**: [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/1814) rebased this pull request as part of a merge.
* **Mar 27, 1:34 PM EDT**: @sgrimm merged this pull request with [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/1814).

**Pull request merged**